nupm/install.nu

```diff
@@ -0,0 +1,98 @@
+use std log
+
+def nupm-home [] {
+    $env.NUPM_HOME? | default (
+        $env.XDG_DATA_HOME?
+        | default ($nu.home-path | path join ".local" "share")
+        | path join "nupm"
+    )
+}
+
+def throw-error [
+    error: string
+    text?: string
+    --span: record<start: int, end: int>
+] {
+    let error = $"(ansi red_bold)($error)(ansi reset)"
+
+    if $span == null {
+        error make --unspanned { msg: $error }
+    }
+
+    error make {
+        msg: $error
+        label: {
+            text: ($text | default "this caused an internal error")
+            start: $span.start
+            end: $span.end
+        }
+    }
+}
+
+def open-package-file [path: path] {
+    let package_file = ($path | path join "package.nuon")
+
+    if not ($package_file | path exists) {
+        throw-error $"package_file_not_found(ansi reset):\nno 'package.nuon' found in ($path)"
+    }
+
+    let package = (open $package_file)
+
+    log debug "checking package file for missing required keys"
+    let missing_keys = (
+        [
+            [key required];
+
+            [$. true]
+            [$.name true]
+            [$.version true]
+            [$.description true]
+            [$.license true]
+        ] | each {|key|
```

Comment:
I think this could be simplified to `keys|all {|key| $key in ($package|columns)}`

Reply:
i've greatly simplified the *package* file key check in 8ab9e69 :yum: 

however, i went with `where` instead of `all` and `columns` because `columns` only returns the list of depth-1 cell-paths :thinking: 
and if we want to test for more complex cell paths... it's more versatile to use `where` and `get -i` :yum: